### PR TITLE
improve read_yaml between osbs-client and atomic-reactor

### PR DESCRIPTION
**Pull request by mlangsdorf**
OSBS-8786

Remove most of the logic in atomic-reactor's read_yaml() and instead call the osbs-client version of the function with 'atomic_reactor' as the package argument.  Keep the existing read_yaml() as an alias to avoid making a bunch of minor changes in the files.

Remove container.json from atomic-reactor so that only one copy will exist in osbs-client.

Requires https://github.com/containerbuildsystem/osbs-client/pull/960

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [ ] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/util.py [hunk 1]
```diff
@@ -1280,48 +1278,15 @@ def read_yaml_from_url(url, schema):
...
     f.seek(0)
-    return read_yaml(f.read(), schema)
+    return osbs_read_yaml(f.read(), schema, package)
 
 
-def read_yaml(yaml_data, schema):
+def read_yaml(yaml_data, schema, package='atomic_reactor'):
     """
     :param yaml_data: string, yaml content
+    :param schema: string, path name to the schema file within the atomic-reactor package
```
Comment (athos-ribeiro) [inline atomic_reactor/util.py]: Could we also decribe 'package' here?

**Comment by mlangsdorf** [comment]
We can't have osbs-client use a schema in atomic-reactor, because that will create a circular dependency.  We don't want to have container.json in two places.  I don't see any other solution than to have it osbs-client and require that changes to container.json get merged there.

File: atomic_reactor/util.py (hunk 1 above)
Comment (mlangsdorf) [reply atomic_reactor/util.py]: Fixed.

**Comment by mlangsdorf** [comment]
Not adding anything to release notes as this is an internal organization change.

**Pull request merged**